multimodal/src/autogluon/multimodal/optimization/losses.py

```diff
@@ -117,3 +135,140 @@ def __init__(self):
     def forward(self, input: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
         loss = torch.sum(-target * F.log_softmax(input, dim=-1), dim=-1)
         return loss.mean()
+
+
+def gather_features(
+    image_features, text_features, local_loss=False, gather_with_grad=False, rank=0, world_size=1, use_horovod=False
+):
+    """
+    Gather features across GPUs.
+
+    Parameters
+    ----------
+    image_features
+        image features of the current process.
+    text_features
+        text features of the current process.
+    local_loss
+        If False, make sure the features on the current GPU have gradients.
+    gather_with_grad
+        Whether to gather all features with gradients enabled.
+    rank
+        Rank of the current process (it should be a number between 0 and world_size-1).
+    world_size
```

Comment:
Just to confirm, the world size here means number of process, not number of GPUs?

Reply:
In GPU machines, they are usually the same. In TPU servers, I think it means TPU cores.